### DISCUSSION TITLE
accumulate csvreader input

### DISCUSSION THREAD

**Discussion by emilyjanedennis** [bonsai-rx/bonsai#1244] [Q&A] [answered]
hiya, attempting to do something that should be easy but isn't clear how to beat go about it... I'd like to use CsvReader and accumulate the rows into a list (or array or whatever) and then use that list/array/whatever in a python transform node... any hints? Scan seems promising but without an example is hard for me to figure out how to use... thanks!

**Comment by bruno-f-cruz** [comment, marked as answer]
Heya,

In general, you can use the `ToArray` operator to create arrays of any type. This operator packages a full sequence (it must be successfully terminated) into a single array. A good practice is then to store this array in some subject with memory. E.g.:
![image](https://user-images.githubusercontent.com/7049351/217478098-a80777c7-349b-45db-b72e-200c45113e3a.png)

Whenever you want to use it, get the value from it and pass it to the next operator:
![image](https://user-images.githubusercontent.com/7049351/217478211-75e82184-1cca-442d-a871-a14f5991466d.png)

Regarding the `PythonTransform`, this will likely change in the new future as we are working on a new Python package for Bonsai, but for now here's an example on how you could manipulate the array:

```python
@returns(int)
def process(value):
  accumulator = 0
  for pair in value:
      accumulator += (pair.Item1 * pair.Item2)
  return accumulator
```
[Arrays_Python.zip](https://github.com/bonsai-rx/bonsai/files/10683478/Arrays_Python.zip)

Hope it helps!
